functions/completed-checks/v1.js

```diff
@@ -0,0 +1,259 @@
+'use strict'
+
+const moment = require('moment')
+const sqlService = require('less-tedious')
+const uuid = require('uuid/v4')
+const { TYPES } = require('tedious')
+
+const config = require('../config')
+sqlService.initialise(config)
+const azureStorageHelper = require('../lib/azure-storage-helper')
+const sqlUtil = require('../lib/sql-helper')
+const markingService = require('./marking.service')
+
+// SQL server vars
+const checkResultTable = '[checkResult]'
+const schema = '[mtc_admin]'
+const checkStatusTable = '[checkStatus]'
+const checkTable = '[check]'
+
+const v1 = {
+  process: async function (context, completedCheckMessage) {
+    await handleCompletedCheck(context, completedCheckMessage)
+  }
+}
+
+// Table Storage
+const azureTableService = azureStorageHelper.getPromisifiedAzureTableService()
+
+async function handleCompletedCheck (context, completedCheckMessage) {
+  context.log('completed-check: message received', completedCheckMessage.checkCode)
+
+  let checkData
+  try {
+    checkData = await sqlUtil.sqlFindCheckWithFormDataByCheckCode(completedCheckMessage.checkCode)
+  } catch (error) {
+    context.log.error(`completed-check: ERROR: failed to retrieve checkData for checkCode: ${completedCheckMessage.checkCode}`)
+    throw error
+  }
+
+  if (!canCompleteCheck(checkData.code)) {
+    const errorMessage = `completed-check: ERROR: check ${completedCheckMessage.checkCode} is not in a correct state to be completed. Current state is ${checkData.code}`
+    context.log.error(errorMessage)
+    throw new Error(errorMessage)
+  }
+
+  try {
+    await savePayloadToAdminDatabase(completedCheckMessage, checkData, context.log)
+  } catch (error) {
+    context.log.error(`completed-check: ERROR: unable to update admin db payload for [${completedCheckMessage.checkCode}]`)
+    throw error
+  }
+
+  try {
+    await updateAdminDatabaseForCheckComplete(completedCheckMessage.checkCode, context.log)
+  } catch (error) {
+    context.log.error(`completed-check: ERROR: unable to update admin db for [${completedCheckMessage.checkCode}]`)
```

Comment:
rather than explain that an error has occured, just log the error message, its much more useful.  we already know its an error entry due to using the error function to log it, and having to dig into whatever catches your throw is just awkward.  please fix this function wide.